src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -226,20 +249,23 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
 
                 if (readContext.IsDeltaOfT)
                 {
-                    IEnumerable<string> structuralProperties = structuredType.StructuralProperties()
+                    IEnumerable<string> structuralProperties = structuredType.StructuredDefinition().Properties()
                         .Select(edmProperty => EdmLibHelpers.GetClrPropertyName(edmProperty, model));
 
+                    PropertyInfo instanceAnnotationProperty = EdmLibHelpers.GetInstanceAnnotationsContainer(
```

Comment:
Maybe name this as `instanceAnnotationsPropertyInfo` like you were naming it in `DeltaDeletedEntityObjectOfT` constructors